include/glow/CodeGen/MemoryAllocator.h

```diff
@@ -27,11 +27,44 @@
 
 namespace glow {
 
+/// Utility function to verify if two contiguous integer intervals overlap. Each
+/// interval is described by a half-open interval [begin, end).
+template <class T>
+inline bool intervalsOverlap(T begin1, T end1, T begin2, T end2) {
+  return (std::max(begin1, begin2) < std::min(end1, end2));
+}
+
+/// Allocation structure which represents a request to allocate or free
+/// a memory region (buffer) within a given MemoryAllocator instance.
+struct Allocation {
+
+  /// Type that should be used as a handle.
+  using Handle = const void *;
+
+  /// Allocation handle which uniquely identifies the buffer to be allocated.
+  /// The provided handle must be unique for each buffer.
+  Handle handle_;
+
+  /// Allocation request type flag: true for ALLOC, false for FREE.
+  bool alloc_;
+
+  /// Allocation buffer size in bytes. This field is mandatory for ALLOC
+  /// and is ignored for FREE.
+  uint64_t size_;
+
+  Allocation(Handle handle, bool alloc, uint64_t size)
+      : handle_(handle), alloc_(alloc), size_(size) {}
+
+  Allocation(size_t id, bool alloc, uint64_t size)
+      : handle_((Handle)(id)), alloc_(alloc), size_(size) {}
+};
+
 /// A POD struct that represents a single half-open allocation [start .. end).
-class Segment {
-public:
+struct Segment {
+
   /// The allocation starts at this address.
   uint64_t begin_;
```

Comment:
See my comment about the naming convention.

Reply:
Done.